sdk/core/Azure.Core.Experimental/src/Primitives/BinaryData.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Abstraction for a payload of binary data.
+    /// </summary>
+    public struct BinaryData
```

Comment:
We should actually make this a [`readonly struct`](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/struct#readonly-struct) since it's immutable.